### DISCUSSION TITLE
How to connect to a VM using Remote Viewer

### DISCUSSION THREAD

**Discussion by rodude123** [cockpit-project/cockpit-machines#1026] [Q&A]
I've installed Cockpit on my Ubuntu machine, I've got cockpit-machines installed as well as set up a VM port-forwarded 5901 5900 outside of the network. I've correctly reversed proxied cockpit through Apache. Now how do I connect to a VM running via Cockpit. If I run `nmap -p 5900` and `nmap -p 5901` to the IP it shows that the ports are closed Also the machine is in a different location to where I'm connecting from

**Comment by guercheLE** [comment]
Have you checked ports are open on firewall? I would also recommend checking ubuntu counterparty to SELinux log (I use fedora server at home)

**Comment by guercheLE** [comment]
> I've installed Cockpit on my Ubuntu machine, I've got cockpit-machines installed as well as set up a VM port-forwarded 5901 5900 outside of the network. I've correctly reversed proxied cockpit through Apache. Now how do I connect to a VM running via Cockpit. If I run `nmap -p 5900` and `nmap -p 5901` to the IP it shows that the ports are closed Also the machine is in a different location to where I'm connecting from

As far as I understand, VMs listen to only 127.0.0.1 (localhost). So any incoming from anything but localhost will be ignored and replies not send back. Are you running nmap from VM's host?

**Comment by rodude123** [comment]
I changed the default ip to the internal IP and now it works
